src/azure-cli/azure/cli/command_modules/sql/_params.py

```diff
@@ -1281,3 +1287,33 @@ def _configure_security_policy_storage_params(arg_ctx):
                    id_part='name')
 
         c.argument('resource_group_name', arg_type=resource_group_name_type)
+
+    ###############################################
+    #             sql instance failover-group     #
+    ###############################################
+
+    with self.argument_context('sql instance-failover-group') as c:
+        c.argument('failover_group_name',
+                   options_list=['--name', '-n'],
+                   help="The name of the Instance Failover Group")
+
+        c.argument('managed_instance',
+                   arg_type=managed_instance_param_type,
+                   options_list=['--source-mi', '--mi'])
+
+        c.argument('partner_managed_instance',
+                   help="The name of the partner managed instance of a Instance Failover Group",
+                   options_list=['--partner-mi'])
+
+        c.argument('partner_resource_group',
+                   help="The name of the resource group of the partner managed instance")
+
+        c.argument('failover_policy',
+                   help="The failover policy of the Instance Failover Group",
+                   arg_type=get_enum_type(FailoverPolicyType))
+
+        c.argument('grace_period',
+                   arg_type=grace_period_param_type)
+
+        c.argument('allow-data-loss',
```

Comment:
does it actually work? there are dashes instead of underscores here

Reply:
yes. I have tests for this.

Reply:
ok @ziwa-msfti trust you :-)
I will merge and probably add another pr with minor updates like this just to make sure it goes in.